tests/unit/lms/services/blackboard_api/client_test.py

```diff
@@ -130,33 +128,34 @@ def test_it_doesnt_send_paginated_requests_forever(
         )
 
     @pytest.mark.parametrize("size,type_", [(1, "File"), (3, "File"), (3, "Folder")])
-    def test_it_emits_files_discover_event(
-        self, svc, basic_client, blackboard_list_files_schema, registry, size, type_
+    def test_it_upserts_files(
+        self,
+        svc,
+        basic_client,
+        blackboard_list_files_schema,
+        file_service,
+        size,
+        type_,
     ):
-        # pylint: disable=protected-access
-        svc._request.registry = registry
         basic_client.request.return_value = factories.requests.Response(json_data={})
         blackboard_list_files_schema.parse.return_value = [
             self.blackboard_file_dict(id_, type_=type_) for id_ in range(size)
         ]
 
         svc.list_files("COURSE_ID")
 
-        svc._request.registry.notify.assert_called_once_with(
-            FilesDiscoveredEvent(
-                request=svc._request,
-                values=[
-                    Any.dict.containing(
-                        {
-                            "lms_id": id_ + 1,
-                            "type": "blackboard_file"
-                            if type_ == "File"
-                            else "blackboard_folder",
-                        }
-                    )
-                    for id_ in range(size)
-                ],
-            )
+        file_service.upsert.assert_called_once_with(
+            [
+                Any.dict.containing(
+                    {
+                        "lms_id": id_ + 1,
+                        "type": "blackboard_file"
+                        if type_ == "File"
+                        else "blackboard_folder",
```

Comment:
I wonder if the `type_` argument to the test could actually be `"blackboard_file"` or `"blackboard_folder"`, then you'd only need `"type": type_` here?